book/plugins/general-abilities/multiple-types-one-repo/multiple-types-one-repo.md

```diff
@@ -0,0 +1,15 @@
+# One Code Base for Multiple Plugins
+
+A developer can open one repository for handling different plugin types.
+
+For example, you can build a push plugin and later add support for an analytic plugin in the same repository.
+
+The key idea for managing such a scenario is "separation of concerns", each plugin type should be implemented on a different class.
+
+Then, you will need to create a different manifest file per plugin type implementation and publish it on Zapp.
```

Comment:
maybe we should also add a few words about that it will also require to add "both" plugins to the app in zapp.

Reply:
In order to do so please do the following steps:
1. Create a separate manifest for each plugin type.
2. Publish them to Zapp while making sure the dependency version matches between manifests.

The differences between these manifests would be:
* Plugin type
* Name and description
* Class name used as an entry point
* Custom parameters

There might be more differences, but those represent the minimal differences that would exist to offer this functionality.

*Note:* It would be beneficial to use a similar / identical versioning scheme between the plugins and mention the relationship in the plugins description.

Reply:
@liviur @snooky23  also the plugin id must be different

Reply:
Done 10x guys!

Reply:
Don't start this sentence with `Then`

This entire sentence is irrelevant with the section you added below this 👇

Reply:
Forgot to remove it.